unstructured/partition/email.py

```diff
@@ -275,12 +275,12 @@ def partition_email(
     metadata_filename: Optional[str] = None,
     metadata_last_modified: Optional[str] = None,
     process_attachments: bool = False,
-    attachment_partitioner: Optional[Callable] = None,
+    attachment_partitioner: Optional[Callable[..., List[Element]]] = None,
```

Comment:
Not that this needs to change, but is there a reason why the ellipse comes before the type?

Reply:
The ellipse indicates the parameters of the callable are not specified (can be any set of parameters). The second type is the return value. The syntax is `Callable[[param-types], return-type]`.

In this case, a partitioner can be just about anything parameter-wise. We could define a callback-protocol that stated things like:
- it has to have a `filename: str` parameter as its first parameter
- it needs to have a `languages: list[str]` parameter somewhere
- etc.

I didn't want to get into that here because I just wanted `json.py` to type-check clean and I don't think a more precise type for a partitioner really buys us much just now. But it's something that might make more sense in the future sometime. A good thing to go in a `model` module.